### ISSUE TITLE
Error importing workflows using n8n CLI

### ISSUE THREAD

**Issue by JavierCane** [n8n-io/n8n#7881]
# TLDR

`n8n import:workflow` CLI command throws a "Cannot read properties of undefined (reading 'on')" error while importing even the most simple workflow from the very same instance to its own, that is, maintaining the same n8n version and config.

# How to reproduce

## Workflow

Workflow with a simple "Manual trigger -> No-op" workflow it is enough to reproduce the error:

![Manual trigger -> No-op](https://github.com/n8n-io/n8n/assets/986235/3657e4aa-e647-4a47-a9e5-1ee769a42cf3)

<details>
  <summary>JSON</summary>
  
```json
{
  "nodes": [
    {
      "parameters": {},
      "id": "bc39234f-9db3-4490-a9f4-8d9d043d4b7d",
      "name": "When clicking \"Execute Workflow\"",
      "type": "n8n-nodes-base.manualTrigger",
      "typeVersion": 1,
      "position": [
        500,
        340
      ]
    },
    {
      "parameters": {},
      "id": "7cb711e5-b697-4f66-adeb-7e85386ade98",
      "name": "No Operation, do nothing",
      "type": "n8n-nodes-base.noOp",
      "typeVersion": 1,
      "position": [
        680,
        340
      ]
    }
  ],
  "connections": {
    "When clicking \"Execute Workflow\"": {
      "main": [
        [
          {
            "node": "No Operation, do nothing",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  },
  "pinData": {}
}
```
</details>

## Commands to execute

- Export the saved workflow: `n8n export:workflow --id=L4KdnXR9ujYHtgZT --output=workflow.json`
- Import it: `n8n import:workflow --input=/home/node/workflow.json`

# Actual behavior

- It can properly export the workflow:
   ```sh
   ~ $ n8n export:workflow --id=L4KdnXR9ujYHtgZT --output=workflow.json
   Successfully exported 1 workflow.
   ```
- But fails while importing it:
   ```sh
   ~ $ n8n import:workflow --input=/home/node/workflow.json
   An error occurred while importing workflows. See log messages for details.
   Cannot read properties of undefined (reading 'on')
   ```

# Expected behavior

The workflow is imported successfully

# Environment

- OS: Ubuntu 22.04.3 LTS
- n8n Version: 1.18.0
- Node.js Version: v18.18.2
- Database system: PostgreSQL
- Operation mode: `queue`

# Related

This might be related with https://github.com/n8n-io/n8n/issues/7258

**Comment by FranciscoKnebel** [comment]
Exact same error occuring to me, using 1.18.0 with PostgreSQL.
Importing credentials worked just fine. Downgrading to a previous release.

**Comment by FranciscoKnebel** [comment]
Update on this: evolving on the importing tools, all credentials had to be manually recreated, same problem as reported here: https://community.n8n.io/t/problem-loading-credentials-after-instance-update-to-1-14-2/32567

**Comment by janober** [comment]
Fix got released with [`n8n@1.21.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.21.0)

**Comment by artick95** [comment]
whats the fix I am still getting the error
An error occurred while importing workflows. See log messages for details. Cannot read properties of undefined (reading 'on')

**Comment by JavierCane** [comment]
> whats the fix I am still getting the error An error occurred while importing workflows. See log messages for details. Cannot read properties of undefined (reading 'on')

As per @janober comment, upgrading to 1.21.0 or greater should fix the issue. However, the latest stable release is still 1.20.0, so I guess it is a matter of time we can upgrade to a stable release including the fix 😊
